apps/aecore/src/aec_peers.erl

```diff
@@ -434,109 +416,25 @@ metrics(#state{peers = Peers, blocked = Blocked,
                            gb_sets:size(Errored)),
     State.
 
-enter_peer(#peer{uri = Uri} = P, Peers) ->
-    gb_trees:enter(hash_uri(Uri), P, Peers).
-
-set_block_flag(Flag, Uri0, #state{peers = Peers,
-                                  blocked = Blocked} = State)
-  when is_boolean(Flag) ->
-    case lookup_peer(Uri0, State) of
-        none ->
-            PrevInBlocked = gb_sets:is_element(Uri0, Blocked),
-            Blocked1 = update_blocked(Flag, Uri0, Blocked),
-            maybe_publish_block(PrevInBlocked =/= Flag, Uri0, Flag),
-            State#state{blocked = Blocked1};
-        {value, _Hash, #peer{uri = Uri} = Peer} ->
-            PrevInBlocked = gb_sets:is_element(Uri, Blocked),
-            PrevBlocked = PrevInBlocked orelse Peer#peer.blocked,
-            NewPeer = Peer#peer{blocked = Flag},
-            Blocked1 = update_blocked(Flag, Uri, Blocked),
-            Peers1 = enter_peer(NewPeer, Peers),
-            maybe_publish_block(PrevBlocked =/= Flag, Uri, Flag),
-            State#state{peers = Peers1,
-                        blocked = Blocked1}
-    end.
-
-update_blocked(true, Uri, Blocked) ->
-    gb_sets:add_element(Uri, Blocked);
-update_blocked(false, Uri, Blocked) ->
-    gb_sets:del_element(Uri, Blocked).
-
-maybe_publish_block(false, _, _) ->
-    ok;
-maybe_publish_block(true, Uri, Flag) ->
-    Msg = if Flag -> blocked;
-             true -> unblocked
-          end,
-    aec_events:publish(peers, {Msg, Uri}).
+enter_peer(#peer{} = P, Peers) ->
+    gb_trees:enter(hash_uri(uri_of_peer(P)), P, Peers).
 
 timestamp() ->
     erlang:system_time(millisecond).
 
--spec log_ping_error(uri()) -> ok.
-log_ping_error(Uri) ->
-    gen_server:cast(?MODULE, {ping_error, uri(Uri), timestamp()}).
+-spec log_ping_error(peer()) -> ok.
+log_ping_error(Peer) ->
+    gen_server:cast(?MODULE, {ping_error, Peer, timestamp()}).
 
--spec log_good_ping(uri()) -> ok.
-log_good_ping(Uri) ->
-    gen_server:cast(?MODULE, {good_ping, uri(Uri), timestamp()}).
+-spec log_good_ping(peer()) -> ok.
+log_good_ping(Peer) ->
+    gen_server:cast(?MODULE, {good_ping, Peer, timestamp()}).
 
--spec hash_uri(uri()) -> binary().
-hash_uri(Uri) when is_list(Uri)->
-    hash_uri(list_to_binary(Uri));
+-spec hash_uri(http_uri:uri()) -> binary().
 hash_uri(Uri) ->
     Hash = crypto:hash(md4,Uri),  %TODO add some random for execution but constant salt, so people can't mess with uri-s to hide on our list.
```

Comment:
Is this comment relevant? (I know it's not yours)

Reply:
Not my comment. I don't understand how one could potentially do any harm with the hash anyway. I left it because I really did not understand it...

Reply:
Me too. :)